### DISCUSSION TITLE
react-spring carousel!

### DISCUSSION THREAD

**Discussion by Emiliano-Bucci** [pmndrs/react-spring#1245] [Show and tell]
Hi guys! I just wanted to share with you a library i've recently release, created on top of `react-spring` and `react-use-gesture`. https://www.npmjs.com/package/react-spring-carousel-js 

I hope that can be useful for anyone and if you have doubts/feedback, please let me know :)

**Comment by TyrGo** [comment]
This is excellent! I was wanting a neat vertical scroll with these features. One trouble I'm having with another library is that what I'm displaying in the slides changes size on click, etc. And I can't figure out how to get the size of the slide to adjust to the changes. Would this work automatically or easy enough with your new library?

**Reply by Emiliano-Bucci** [reply]
@TyrGo Hi! Do you have an example of your issue?

**Reply by TyrGo** [reply]
Thank you! Here's a simple fragment:

       <div onClick={clickHandler}>

          <div className={orig ? "fadeIn" : "fadeOut"}>

            {orig ? original : null}

          </div>

          <div className={!orig ? "fadeIn" : "fadeOut"}>

            {orig ? null : expanded}

          </div>

        </div>

 And styling:
 
.fadeOut{
     opacity:0;
     transition: opacity 2s;
     transition-delay: 0.2s;
}
.fadeIn{
     opacity:1;
     transition: opacity 2s;
     transition-delay: 0.2s;
}

I'm not at all sure this is the neatest way to do what I want. But what I want is for the the content of a slide to be replaced on a click. But the new content is bigger, and the libraries I've tried so far can't handle the sizing and transition well. I tried a few minutes ago to see whether yours would, and I'm just stuck on something simple: e.g. 'GridWrapper', 'ArrowButton', etc, seem to be components I should import from your library. But that doesn't seem to work.

**Reply by Emiliano-Bucci** [reply]
Regarding `GridWrapper` or `ArrowButton`, they are custom components created just for example purpose and they don't exist in the library. I've done this in this way so everyone can bring their own components using the desired style system without having to install further dependencies. A part from some basic styles in order to make the carousel work (which i'm inlining without using any css library), i don't provide any kind of components, but just the API to make the carouse work.

**Reply by TyrGo** [reply]
Ah! I'll try it out more carefully then. Do you foresee any trouble with what I'm trying to do?

**Reply by TyrGo** [reply]
Your styling is very elegant, btw. Would be neat if you did include those components too!

**Reply by Emiliano-Bucci** [reply]
Thanks! But i prefer to not add this kind of stuff in the library as my only concern is the functionality of the carousel; how it looks it depends on the developer who implement it. 

Regarding your issue, maybe i should try using a real example but probably you could find some issues, specially regarding the calculation of some dimensions and distances, while from transition perspective i guess you shouldn't have particular issues. I've tried to make it in a way that every calculation is done in a way that possible changes wouldn't affect, but your case is a bit particular, as normally, from a UX perspective, the size of the carousel don't change. In case, if you provide me with a simple example of your case (just an example with mocked data) i would gladly take a look and maybe i could help you. Let me know :)

**Reply by TyrGo** [reply]
Thank you! I think I'll try to figure it out with your library over the next couple days, and if/when I have trouble, I'll send along a little sample and would be grateful for any advice! Best wishes, Ty
